src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -219,8 +219,11 @@ function VisEditor($rootScope, $scope, $route, timefilter, AppState, $window, kb
       return savedVis.lastSavedTitle || `${savedVis.title} (unsaved)`;
     };
 
-    $scope.$watch('searchSource.get("index").timeFieldName', function (timeField) {
-      timefilter.enabled = !!timeField;
+    $scope.$watchMulti([
+      'searchSource.get("index").timeFieldName',
+      'vis.type.requiresTimePicker',
+    ], function ([timeField, requiresTimePicker]) {
+      timefilter.enabled = Boolean(timeField || requiresTimePicker);
```

Comment:
I can't think of a reason why a visualization would use an index pattern with a `timeField` but not want a Timepicker? Do you have a use case in mind? My only objection to adding that feature (if we don't have a use case in mind) is it would add more logic that may not need to be there.